src/daemon/messages.go

```diff
@@ -269,6 +271,23 @@ func (intro *IntroductionMessage) Handle(mc *gnet.MessageContext, daemon interfa
 
 		logger.Infof("%s verified for version %d", mc.Addr, intro.Version)
 
+		// Checks the genesis hash if not empty
+		if len(intro.Extra) > 0 {
+			var bcPubKey cipher.PubKey
+			if len(intro.Extra) != len(bcPubKey) {
```

Comment:
IMO it would be nice to have a typed struct instead of an opaque `Extra byte[]` .

Reply:
Its transitory, next version will have a defined struct